File: perl/examples/sort_dates.pl
#!/usr/bin/perl -w

sub random_date {
	return sprintf "%02d/%02d/%04d", 1 + rand 28, 1 + rand 12, 2000+rand 20
}

sub compare_date {
	my ($day1,$month1,$year1) = split /\D+/, $a;
	my ($day2,$month2,$year2) = split /\D+/, $b;
	return $year1 <=> $year2 || $month1 <=> $month2 || $day1 <=> $day2;
}

push @random_dates, random_date() foreach 1..5;
print "random_dates=@random_dates\n";
@sorted_dates = sort compare_date @random_dates;
print "sorted dates=@sorted_dates\n";

File: python/examples/sort_dates.py
#!/usr/bin/python
import re
from random import randint
def random_date():
	return "%02d/%02d/%04d"%(randint(1,28), randint(1,12), randint(2000,2020))

def compare_date(date1, date2):
	(day1,month1,year1) = re.split(r'\D+', date1)
	(day2,month2,year2) = re.split(r'\D+', date2)
	return cmp(year1,year2) or cmp(month1,month2) or cmp(day1,day2)

random_dates = [random_date() for x in xrange(0,5)]
print "random_dates=",",".join(random_dates)
sorted_dates = sorted(random_dates, cmp=compare_date)
print "sorted dates=",",".join(sorted_dates)